packages/grid/x-data-grid-premium/src/components/GridColumnMenuAggregationItem.tsx

```diff
@@ -74,32 +90,36 @@ function GridColumnMenuAggregationItem(props: GridColumnMenuItemProps) {
         <rootProps.slots.columnMenuAggregationIcon fontSize="small" />
       </ListItemIcon>
       <ListItemText>
-        <FormControl size="small" fullWidth sx={{ minWidth: 150 }}>
+        <rootProps.slots.baseFormControl size="small" fullWidth>
           <InputLabel id={`${id}-label`}>{label}</InputLabel>
-          <Select
+          <rootProps.slots.baseSelect
+            aria-labelledby={`${id}-label`}
             labelId={`${id}-label`}
-            id={`${id}-input`}
             value={selectedAggregationRule}
             label={label}
             color="primary"
             onChange={handleAggregationItemChange}
-            onBlur={(e) => e.stopPropagation()}
+            onBlur={(e: FocusEvent) => e.stopPropagation()}
             fullWidth
           >
-            <MenuItem value="">...</MenuItem>
             {availableAggregationFunctions.map((aggFunc) => (
-              <MenuItem key={aggFunc} value={aggFunc}>
+              <rootProps.slots.baseSelectOption key={aggFunc} value={aggFunc}>
                 {getAggregationFunctionLabel({
                   apiRef,
                   aggregationRule: {
                     aggregationFunctionName: aggFunc,
                     aggregationFunction: rootProps.aggregationFunctions[aggFunc],
                   },
                 })}
-              </MenuItem>
+              </rootProps.slots.baseSelectOption>
             ))}
-          </Select>
-        </FormControl>
+          </rootProps.slots.baseSelect>
+          {selectedAggregationRule ? (
+            <rootProps.slots.baseIconButton onClick={handleAggregationItemChange}>
+              <rootProps.slots.columnMenuClearIcon fontSize="small" color="action" />
+            </rootProps.slots.baseIconButton>
+          ) : null}
```

Comment:
Should the clear icon be part of the text field? It sounds more like an external element, it's rendered outside of the input.

Reply:
Yes, making it part of the text field would make it feel a bit better. `endAdornment` worked but by default it collided with the dropdown icon, so I had to add a margin, is it a visual bug 🤔 
<img width="249" alt="image" src="https://github.com/mui/mui-x/assets/12609561/5e56c6c3-1775-4897-88a7-0d7fc3cf25c6">

> I wonder if we wouldn't want to move toward removing baseFormControl, baseInputLabel, baseSelect as a slot. This concept feels MUI-centered, not something that translates often to custom design systems that developers might want to customize the data grid with. Instead, we could replace them with an input and select slots (not sure, but could).

Do you mean native input/select components? I feel currently we are going in the other direction, by adding slots for all possible components for them to be able to be swapped between Material/Joy UI/any future design system's components. I can't picture how a developer will add their custom design system on top if we replace these items with the native input/select. 
Or the point is to make it custom-interface-free (support the native props only) so that it's compatible with most users' use cases?